modules/twinkleprotect.js

```diff
@@ -1114,28 +1114,19 @@ Twinkle.protect.callback.changePreset = function twinkleprotectCallbackChangePre
 
 Twinkle.protect.callback.evaluate = function twinkleprotectCallbackEvaluate(e) {
 	var form = e.target;
-
-	var actiontypes = form.actiontype;
-	var actiontype;
-	for (var i = 0; i < actiontypes.length; i++) {
-		if (!actiontypes[i].checked) {
-			continue;
-		}
-		actiontype = actiontypes[i].values;
-		break;
-	}
+	var input = Morebits.quickForm.getInputData(form);
 
 	var tagparams;
-	if (actiontype === 'tag' || (actiontype === 'protect' && mw.config.get('wgArticleId') && mw.config.get('wgPageContentModel') !== 'Scribunto')) {
+	if (input.actiontype === 'tag' || (input.actiontype === 'protect' && mw.config.get('wgArticleId') && mw.config.get('wgPageContentModel') !== 'Scribunto')) {
 		tagparams = {
-			tag: form.tagtype.value,
-			reason: (form.tagtype.value === 'pp-protected' || form.tagtype.value === 'pp-semi-protected' || form.tagtype.value === 'pp-move') && form.protectReason ? form.protectReason.value : null,
-			small: form.small.checked,
-			noinclude: form.noinclude.checked
+			tag: input.tagtype,
+			reason: (input.tagtype === 'pp-protected' || input.tagtype === 'pp-semi-protected' || input.tagtype === 'pp-move') && (input.protectReason || null),
```

Comment:
Not sure about this. On line 1499 (where `tagparams` is now called `params`, we have 
```js
		newtag += "'''" + Morebits.string.toUpperCaseFirstChar(words) + (params.reason !== '' ? ":''' " +
			Morebits.string.formatReasonText(params.reason) : ".'''") + ' ~~~~';
```
which looks buggy? (`params.reason` would never be `''` it would be null).

Reply:
That line (`newtag`, 1499) is in `fileRequest`, which doesn't actually use `tagparams` directly, it creates `rppparams`.  In `rppparams`, `reasons` is created by first setting `typereason` to the text for whatever type of protection was chosen to request, with the `default` (1357) as `typereason = '';`.  From there, `reason = typereason;`, so `reason` isn't ever null.  in 1363 we check `input.reason !== ''`, so no, I don't think `null` comes into it here (just confusing code).

Reply:
Ok done, as you seem to know what's going on!

Reply:
That is a big assumption! 😆